pkg/controllers/podfacts.go

```diff
@@ -271,6 +278,21 @@ func (p *PodFacts) checkIfNodeIsUp(ctx context.Context, pf *PodFact) error {
 	return nil
 }
 
+// checkIfAgentRunning will check if the Vertica agent is running and set state in pf.agentRunning
+func (p *PodFacts) checkIfAgentRunning(ctx context.Context, pf *PodFact) error {
+	if !pf.isPodRunning {
+		pf.agentRunning = false
+		return nil
+	}
+
+	if _, _, err := p.PRunner.ExecInPod(ctx, pf.name, ServerContainer, "/opt/vertica/sbin/vertica_agent", "status"); err != nil {
+		pf.agentRunning = false
+	} else {
```

Comment:
Isn't there a way to avoid this if else? I know it does not change anything.

Reply:
Yeah, makes sense.  I can clean this up.